web/modules/custom/hello_user/src/Controller/HelloUser.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Drupal\hello_user\Controller;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\hello_user\CurrentUser;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Returns responses for hello user content routes.
+ */
+class HelloUser extends ControllerBase {
+  /**
+   * The current user.
+   * 
+   * @var \Drupal\hello_user\CurrentUser
+   */
+  protected $currentUser;
+  /**
+   * The content to be rendered.
+   * 
+   * @var array
+   */
+  protected $content = [];
+
+  /**
+   * Constructs new HelloUser controller object.
+   * 
+   * @param \Drupal\hello_user\CurrentUser $currentUser
+   *   The current user service.
+   */
+  public function __construct(CurrentUser $currentUser) {
+    $this->currentUser = $currentUser;
+  }
+  
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('hello_user.current_user'),
+    );
+  }
+  
+  /**
+   * Renders a page to view user name.
+   * 
+   * @return array
+   *   An array suitable for showing content.
+   */
+  public function view() {
+    $this->content['name'] = $this->currentUser->getUserDisplayName();
```

Comment:
1/ Why assign this value to a variable?
2/ Why create a class property just for this?

Reply:
1/ Has assigned value to variable to test passing of more than one values to template. Removed now.
2/ Changed